### PR TITLE
RT: Use CPT page and mShots for previewing RTs

### PR DESCRIPTION
**Pull request by mmtr**
Fixes https://github.com/Automattic/dotcom-forge/issues/8617
Requires D157805-code

## Proposed Changes

- Moves away from the block renderer endpoint to generate RT previews because we're not able to simulate the look&feel of the frontend in there
- Instead we now preview the frontend of the RT pages and mShot them

Before | After
--- | ---
Screenshot 2024-08-08 at 11 59 18 | Screenshot 2024-08-08 at 13 33 50

- Note: I initially used the `WebPreview` component but had to abandon it because [Chrome has subpixel rendering issues when scaling the content with CSS](https://issues.chromium.org/issues/323982813). It's easier to just use mShots because there are no rendering issues when scaling down images.
- Note 2: Because we have slugs now (needed for the previews), I thought it'd be interesting to use them in the URLs instead of the IDs.

## Why are these changes being made?

Because the current RT previews don't reflect the actual layout applied to the site

## Testing Instructions

- Use the Calypso live link below
- Go to `/patterns`
- Scroll down to the RT section
- Make sure the preview of Cafe looks good now (no horizontal line between header and content, width of the drinks menu is wider)
- Open the RT details
- Make sure the preview still looks good

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: client/my-sites/patterns/components/readymade-template-preview/index.tsx [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+	const mShotsOptions = {
+		vpw: 1200,
+		vph: 1680,
+		w: 1200,
+		screen_height: 1680,
+	};
+
+	return (
+		<div className="readymade-template-preview">
+			<MShotsImage url={ readymadeTemplate.previewUrl + '&t=2' } alt="" options={ mShotsOptions } />
```
Comment (mmtr) [inline client/my-sites/patterns/components/readymade-template-preview/index.tsx]: The `t=2` param here is intentional. Once D157805-code is shipped, I'll remove it so we can easily generate new mShots

**Review by dsas — COMMENTED** [review]
It feels like the previews are slightly slower to render but I think this is ok.

Previews that are half off of the screen don't always get rendered until they're fully on it. 
Screenshot 2024-08-08 at 12 30 23

It might be a bit finicky to reproduce, it seems like 

1. With an empty cache go to /patterns#readymade-templates-section
2. Use the carousel buttons so that General contractor and Bakery are in the middle and spa is half off the left edge
3. Reload the page

I find myself advanced through the carousel to the same place as before, but instead of seeing half of the spa preview, I get a white box

**Comment by mmtr** [comment]
> Previews that are half off of the screen don't always get rendered until they're fully on it.

I think that's an effect of the lazy loading we do for mShots images:

https://github.com/Automattic/wp-calypso/blob/92c289eeb28b00630c94316f45643c10bf16a31f/packages/onboarding/src/mshots-image/index.tsx#L281

See https://developer.mozilla.org/en-US/docs/Web/Performance/Lazy_loading

> The [loading](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/img#loading) attribute on an [](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/img) element, can be used to instruct the browser to defer loading of images/iframes that are off-screen until the user scrolls near them.

**Review by dsas — APPROVED** [review]
> I think that's an effect of the lazy loading we do for mShots images:

Yeah it's probably that. Maybe it's a problem with Firefox's lazy loading behaviour. I can't reproduce in Chrome - when I reload in Chrome the carousel is rewound to the beginning.

It ought to be fixed, but I don't think it's a blocker.

**Comment by mmtr** [comment]
> Maybe it's a problem with Firefox's lazy loading behaviour.

I avoided the lazy behavior in https://github.com/Automattic/wp-calypso/pull/93368/commits/c8a71b9da485f1a73429e575a63f0d8d28d43677

**Pull request merged**